packages/cloud_firestore/cloud_firestore_platform_interface/lib/src/platform_interface/platform_interface_load_bundle_task_snapshot.dart

```diff
@@ -0,0 +1,41 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+import '../../cloud_firestore_platform_interface.dart';
+
+/// The interface a load bundle task snapshot must extend.
+class LoadBundleTaskSnapshotPlatform extends PlatformInterface {
+  // ignore: public_member_api_docs
+  LoadBundleTaskSnapshotPlatform(this.taskState, this._data)
+      : super(token: _token);
+
+  static final Object _token = Object();
+
+  final Map<String, dynamic> _data;
```

Comment:
Instead of storing a map and having getters, deserialize the map in the constructor.